PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskEA.cxx

```diff
@@ -0,0 +1,825 @@
+
+  TString trigger = fInputEvent->GetFiredTriggerClasses();
+  bool passedTrigger = kFALSE;
+  if(trigger.Contains("INT7")){
+     passedTrigger = kTRUE;
+  }
+  return passedTrigger;
+
+}
+
+//_____________________________________________________________________________________
+Double_t AliAnalysisTaskEA::GetExternalRho(Bool_t isMC){
+
+   // Get rho from event using CMS approach
+   AliRhoParameter* rho = NULL;
+   TString rhoname = (!isMC) ? fRhoTaskName : fRhoTaskNameMC;
+   if(!rhoname.IsNull()){
+      rho = dynamic_cast<AliRhoParameter*>(InputEvent()->FindListObject(rhoname.Data()));
+      if (!rho) {
+        //AliWarningF(MSGWARNING("%s: Could not retrieve rho with name %s!"), GetName(), rhoname.Data());
+        return 0.;
+      }
+   }else{
+      //AliWarningF(MSGWARNING("No %s Rho task name provided"), (!isMC ? "DATA" : "MC"));
+      return 0.;
+   }
+   
+   return rho->GetVal();
+}
+//________________________________________________________________________
+
+Bool_t AliAnalysisTaskEA::IsEventInAcceptance(AliVEvent* event){
+   //EVENT SELECTION RECONSTRUCTED DATA
+
+
+   if(!event) return kFALSE;
+
+   //___________________________________________________
+   //TEST PILE UP
+   if(fUsePileUpCut){
+      if(!fHelperClass || fHelperClass->IsPileUpEvent(event)){ 
+         fHistEvtSelection->Fill(1.5); //count events rejected by pileup
+         return kFALSE;
+      }
+   }
+   //___________________________________________________
+   //VERTEX CUT
+
+   if(fUseDefaultVertexCut){
+      if(!fHelperClass || !fHelperClass->IsVertexSelected2013pA(event)){  //??? USE THIS OR SOMETHING ELSE
+         fHistEvtSelection->Fill(2.5); //count events rejected by vertex cut 
+         return kFALSE;
+      }
+   }else{
+      if(TMath::Abs(event->GetPrimaryVertex()->GetZ()) > fZVertexCut){
+         fHistEvtSelection->Fill(2.5); //count events rejected by vertex cut 
+         return kFALSE;
+      }
+   }
+   //___________________________________________________
+   //AFTER VERTEX CUT
+   fhVertexZ->Fill(event->GetPrimaryVertex()->GetZ()); 
+
+
+  return kTRUE;
+}
+
+//________________________________________________________________________
+Bool_t AliAnalysisTaskEA::IsTrackInAcceptance(AliVParticle* track, Bool_t isGen){
+   // Check if the track pt and eta range 
+   if(!track) return kFALSE;
+
+   if(isGen){ //particle level MC:   select charged physical primary tracks 
+      //Apply only for kine level or MC containers   
+      if(!track->Charge()) return kFALSE;
+      if(fMC == kPartLevel){
+         if(!(static_cast<AliAODMCParticle*>(track))->IsPhysicalPrimary()) return kFALSE;
+      }    
+   }
+   if(TMath::Abs(track->Eta()) < fTrackEtaWindow){ //APPLY TRACK ETA CUT
+      if(track->Pt() > fMinTrackPt){   //APPLY TRACK PT CUT
+         return kTRUE;
+      }
+   }
+   return kFALSE;
+}
+//________________________________________________________________________
+void AliAnalysisTaskEA::ExecOnceLocal(){
```

Comment:
AliAnalysisTaskEmcal provides an interface UserExecOnce which you could have used for this.